src/devices/Display/Large4Digit14SegmentDisplay.cs

```diff
@@ -0,0 +1,285 @@
+        {
+            0 => (int)Address.Digit1,
+            1 => (int)Address.Digit2,
+            2 => (int)Address.Digit3,
+            3 => (int)Address.Digit4,
+            _ => throw new ArgumentOutOfRangeException(nameof(digit)),
+        };
+        #endregion
+
+        #region Public members
+
+        /// <summary>
+        /// Initialize 7-Segment display
+        /// </summary>
+        /// <param name="i2cDevice">The <see cref="I2cDevice"/> to create with.</param>
+        public Large4Digit14SegmentDisplay(I2cDevice i2cDevice)
+            : base(i2cDevice)
+        {
+        }
+
+        /// <inheritdoc/>
+        public int NumberOfDigits { get; } = MaxNumberOfDigits;
+
+        /// <summary>
+        /// Gets or sets a single digit's segments by id
+        /// </summary>
+        /// <param name="address">digit address (0..3)</param>
+        /// <returns>Segment in display buffer for the given address</returns>
+        public Segment14 this[int address]
+        {
+            get => (Segment14)_displayBuffer[TranslateDigitToBufferAddress(address)];
+            set
+            {
+                _displayBuffer[TranslateDigitToBufferAddress(address)] = (byte)value;
+                AutoFlush();
+            }
+        }
+
+        /// <inheritdoc/>
+        /// <remarks>Write clears dots, you'll have to reset them afterwards</remarks>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="startAddress"/></exception>
+        /// <exception cref="ArgumentOutOfRangeException"><see cref="Large4Digit7SegmentDisplay"/> only supports <see cref="MaxNumberOfDigits"/> digits</exception>
+        public override void Write(ReadOnlySpan<byte> digits, int startAddress = 0)
+        {
+            Console.WriteLine($"Length: {digits.Length}");
+            if (digits.Length == 0)
+            {
+                // Nothing to write
+                return;
+            }
+
+            if (startAddress < 0 || startAddress >= (MaxNumberOfDigits * 2))
+            {
+                throw new ArgumentOutOfRangeException(nameof(startAddress));
+            }
+
+            if (digits.Length + startAddress > (MaxNumberOfDigits * 2))
+            {
+                throw new ArgumentOutOfRangeException(nameof(digits), $"{nameof(Large4Digit7SegmentDisplay)} only supports {MaxNumberOfDigits - startAddress} digits starting from address {startAddress}");
+            }
+
+            foreach (byte digit in digits)
+            {
+                _displayBuffer[(int)s_digitAddressList[startAddress++]] = (byte)(digit);
+            }
+
+            AutoFlush();
+        }
+
+        /// <summary>
+        /// Write raw data to display buffer
+        /// </summary>
+        /// <param name="digits">Array of ushort to write to the display</param>
+        /// <param name="startAddress">Address to start writing from</param>
+        public void Write(ReadOnlySpan<ushort> digits, int startAddress = 0)
```

Comment:
Please make sure members are only public if they need to be. This one doesn't look like it's useful from outside.

Reply:
Done